lib/pbench/server/database/models/users.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+from flask_bcrypt import generate_password_hash
+from email_validator import validate_email
+from sqlalchemy import Column, Integer, String, DateTime, LargeBinary
+from pbench.server.database.database import Database
+from sqlalchemy.orm import relationship, validates
+
+
+class User(Database.Base):
+    """ User Model for storing user related details """
+
+    __tablename__ = "users"
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    username = Column(String(255), unique=True, nullable=False)
+    first_name = Column(String(255), unique=False, nullable=False)
+    last_name = Column(String(255), unique=False, nullable=False)
+    password = Column(LargeBinary(128), nullable=False)
+    registered_on = Column(DateTime, nullable=False, default=datetime.datetime.now())
+    email = Column(String(255), unique=True, nullable=False)
+    auth_tokens = relationship("ActiveTokens", backref="users")
+
+    def __str__(self):
+        return f"User, id: {self.id}, username: {self.username}"
+
+    def get_json(self):
+        return {
+            "username": self.username,
+            "email": self.email,
+            "first_name": self.first_name,
+            "last_name": self.last_name,
+            "registered_on": self.registered_on,
+        }
+
+    @staticmethod
+    def get_protected():
+        return ["registered_on", "id"]
```

Comment:
That's an interesting question; nothing but token add/remove should be altering the user's `auth_tokens` list. We always deliver sanitized JSON without `auth_tokens` to the `GET` client, and we shouldn't allow `auth_tokens` to creep in on a `PUT`.

Reply:
Well this is an interesting question, we internally do modify auth tokens associated with a user but when a user re-login. However, we definitely need to add an interface to avoid end-user trying to modify the auth tokens, I will create an issue to track

Reply:
Created an issue https://github.com/distributed-system-analysis/pbench/issues/2127